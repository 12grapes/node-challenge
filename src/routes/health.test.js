const request = require('supertest')
const server = require('../app').listen()

// Close the server after each test
afterEach(() => {
  server.close()
})

describe('/health', () => {
  it('should respond 200 and JSON with an success: true status', async () => {
    const response = await request(server).get('/health')
    expect(response.status).toEqual(200)
    expect(response.type).toEqual('application/json')
    expect(response.body).toMatchSnapshot()
  })
  it('should log the request in elasticsearch', async () => {
    const response = await request(server).get('/health')
    expect(response.status).toEqual(200)
    expect(response.type).toEqual('application/json')
    // TODO: add more assertions
  })
})

describe('/weekly-report', () => {
  // TODO: write tests
})
