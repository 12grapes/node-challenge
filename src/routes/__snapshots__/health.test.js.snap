// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/health should respond 200 and JSON with an success: true status 1`] = `
Object {
  "result": Object {
    "result": "created",
  },
  "success": true,
}
`;
